packages/react/src/components/Slider/Slider.js

```diff
@@ -205,6 +209,28 @@ export default class Slider extends PureComponent {
       // Reset the flag
       this.setState({ needsOnRelease: false });
     }
+
+    if (
+      useControlledStateWithValue &&
+      this.props.value === '' &&
+      prevState.value !== ''
+    ) {
+      this.setState({ value: '' });
+    }
+
+    // If `useControlledStateWithValue` feature flag is on, do nothing here.
+    // Otherwise, do prop -> state sync without "value capping".
+    if (useControlledStateWithValue || prevState.value === this.props.value) {
```

Comment:
We need to perform change detection in props, like what `useEffect()` does with its 2nd arg:

```suggestion
    if (useControlledStateWithValue || prevProps.value === this.props.value) {
```